dnn/generated_algorithmdescriptor.go

```diff
@@ -10,18 +10,37 @@ import "runtime"
 type AlgorithmDescriptor struct {
 	internal C.cudnnAlgorithmDescriptor_t
 
-	algorithm TODO
+	algorithm interface{}
```

Comment:
Should we name this interface, and give dummy methods to things that implement it? Thoughts?

Reply:
It's good to hide the type in AlgorithmDescriptor, so in future we can use union replacement if any :)